sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs

```diff
@@ -30,7 +30,7 @@ public class ProcessSessionMessageEventArgs : EventArgs
         /// </summary>
         public CancellationToken CancellationToken { get; }
 
-        internal ConcurrentDictionary<ServiceBusReceivedMessage, byte> Messages => _receiveActions.Messages;
+        internal ConcurrentDictionary<ServiceBusReceivedMessage, CancellationTokenSource> Messages => _receiveActions.Messages;
```

Comment:
How are we managing disposal of the CTS?   I'm not sure if I overlooked it or you haven't gotten there yet.  😄

Reply:
It's here https://github.com/Azure/azure-sdk-for-net/pull/36253/files#diff-2886e564cebcf9a5af7ef07e32dbc090dab7bb95304ef0b3d8ab4b5fedfe9355R155-R159

`CancelMessageLockRenewalAsync` is always called in the finally regardless of the mode the processor operates in.